### PR TITLE
Fixes #48 renameAttributeMeta does not exist on client API

### LINKED ISSUE #48: renameAttributeMeta does not exist on client API

### LINKED ISSUE #49: Core should check for valid attribute, pointer, aspects and set names.
Would be nicer if a synchronous exception was thrown here rather than waiting till mongo complains about the '.' and '$' characters. There is already a check for the name.

### LINKED ISSUE #51: Core.isValidAttributeValueOf should return false if attribute is readonly and value holder not a metanode

### PR DESCRIPTION
**Pull request by kecso**
Fixes #49 Core should check for valid attribute, pointer, aspects and set names.
Fixes #51 Core.isValidAttributeValueOf should return false if attribute is readonly and value holder not a metanode

### CODE REVIEW

File: src/common/core/core.js [hunk 1]
```diff
@@ -3488,7 +3524,8 @@ define([
         this.addLibrary = function (node, name, libraryRootHash, libraryInfo, callback) {
             ensureType(callback, 'callback', 'function');
             var error = ensureNode(node, 'node', true);
-            error = error || ensureType(name, 'name', 'string');
+            error = error || ensureType(name, 'name', 'string', true);
+            error = error || ensureMongoCompatibleKey(name, 'name', false, true);
```
Comment (pmeijer) [inline src/common/core/core.js]: Is it really needed for libraries?

File: src/common/core/core.js [hunk 2]
```diff
@@ -1411,7 +1437,7 @@ define([
         this.createSet = function (node, name) {
             ensureNode(node, 'node');
             ensureRelationName(name, 'name');
-
+            ensureMongoCompatibleKey(name, 'name', false);
```
Comment (pmeijer) [inline src/common/core/core.js]: Why is the `hiddenIsFine` set to false for this an not the others?

File: src/common/core/core.js [hunk 3]
```diff
@@ -287,6 +287,29 @@ define([
...
+            realInput = input;
+
+        if (hiddenIsFine === true && input[0] === '_') {
+            realInput = input.substr(1);
+        }
+
+        if (REGEXP.DOCUMENT_KEY.test(realInput) === false) {
+            error = new CoreIllegalArgumentError('Parameter ' + nameOfInput +
+                ' is not a valid key (cannot contain "." or "$"' +
+                (hiddenIsFine ? '' : 'and starts with "_"') + ').');
```
Comment (pmeijer) [inline src/common/core/core.js]: This sentence is not grammatically correct if `hiddenIsFine == false`.

File: src/common/core/core.js (hunk 1 above)
Comment (kecso) [reply src/common/core/core.js]: For libraries, most of the things (like '.') were already checked, but it made things more consistent. Also, it is necessary as the name of the library will be the name attribute of the root node of the library subtree

Comment (pmeijer) [reply src/common/core/core.js]: :+1:

File: src/common/core/core.js (hunk 2 above)
Comment (kecso) [reply src/common/core/core.js]: We do not allow 'hidden' names for sets, and the underscore is already being tested in the ensureRelationName.
On the other hand I will check to make all set related functions and after ensureRelationName checks consistent.

File: src/common/core/core.js (hunk 3 above)
Comment (kecso) [reply src/common/core/core.js]: True.

**Pull request merged**